Implement MutablePrimitiveList.collectPrimitive().

bhav0904 opened the pull request:
Signed-off-by: Bhavana Hindupur bhavana.hindupur@gmail.com

The intention is to achieve PrimitiveIterable.collectPrimitive() for all 8 primitives. This is the first step.

bhav0904 commented:
Any comments on this?

itohro commented on unit-tests/src/test/java/org/eclipse/collections/impl/list/mutable/primitive/AbstractBooleanListTestCase.java:
```diff
@@ -396,4 +403,91 @@ public void toImmutable()
+        MutableBooleanList list = this.newWith(true, false, true);
+
+        MutableByteList mutableByteList =
+                list.collectByte(each -> each ? (byte) 1 : (byte) 0);
```
Seems this line can be simplified with
`list.collectByte(each -> each ? (byte) (each ? 1 : 0));`

And the same case for `short` and `char` below.

itohro commented:
Besides the one I commented, the changes look good to me overall.

Could you please briefly comment how you envision the steps to complete PrimitiveIterable.collectPrimitive()?

itohro commented:
And please add a draft release note for this change, too.

bhav0904 commented:
Hi Hiroshi, Sorry for the delay. I made the following changes:

1. Removed the changes to the interface primitiveArrayList, since this is now targeted for a minor release, 8.1.0.
2. Added a draft release note for the change.
3. Rebased onto master.

The plan for this change is to iteratively add collect<primitive> methods to all implementations in the following order:
1. Primitive lists
2. Primitive sets
3. Primitive maps
4. Primitive stacks
5. Primitive bags
6. (any other collection type I might have missed)

Depending on whether we're targeting a minor or major release, we can choose to add the method to the interface. For e.g., if this change goes in for 8.1.0, I will add the collect<Primitive> methods to the primitiveList interface for 9.0.

gs-rezaem commented:
Please rebase and make sure the builds are green.

bhav0904 commented:
Sure Moh, sorry about that. I was having some issues with my maven build locally, which I hopefully will sort out today. I will fix the builds.

bhav0904 commented:
I tried resolving a merge conflict using the github UI, and the commit was made with the email ID: noreply@github.com. I tried rebasing, but the merge commit isn't coming up in the rebase list to squash it or get rid of it. Can someone help me?

nikhilnanivadekar commented:
@bhav0904 your IP validation is also failing, please make sure you sign-off your commits.

Regarding the merge conflict look at the [network](https://github.com/eclipse/eclipse-collections/network). You need to rebase on top of this [commit](https://github.com/eclipse/eclipse-collections/commit/b18a0d60ea600e37f3afa7a4fed151283723fab8). Try to do [git pull](https://git-scm.com/docs/git-pull) and then rebase.

bhav0904 commented:
I couldnt figure out how to fix this, so I created another pull request. I am closing this one.
